### DISCUSSION TITLE
fs.mv like command line mv

### DISCUSSION THREAD

**Discussion by kapouer** [nodejs/help#4662] [Ideas]
This is not like rename(2) because it would work across boundaries (EXDEV: cross-device link not permitted, rename).
It would also accept a "force" option for overwriting.

**Comment by avivkeller** [comment]
Could you elaborate?

**Comment by kapouer** [comment]
`rename` has a somewhat unexpected contract: it is atomic but it doesn't work across file systems.
A very easy way to realize this is to rename from or to `/tmp` on a modern linux distribution.
This goes against portability.
99.99% of the time, the programmer using fs.rename doesn't care and doesn't even know about that limitation.
However the fs.rename function behavior cannot be changed since it would break atomicity.

Introducing a `fs.mv`, following the similar names of `fs.cp` and `fs.rm` would complete the API nicely.

Note that for `fs.mv` for *files* is really easy to implement, but it is less obvious (at least, to me) for directories.

**Comment by targos** [comment]
This SGTM in principle. @nodejs/fs wdyt?
